Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -0,0 +1,299 @@
+            List<CloudTask> taskCollection = client.getTaskOperations().listTasks(jobId, new DetailLevel.Builder().selectClause("id, state").build());
+
+            boolean allComplete = true;
+            for (CloudTask task : taskCollection) {
+                if (task.getState() != TaskState.COMPLETED) {
+                    allComplete = false;
+                    break;
+                }
+            }
+
+            if (allComplete) {
+                // All tasks completed
+                return true;
+            }
+
+            System.out.println("wait 10 seconds for tasks to complete...");
+
+            // Check again after 10 seconds
+            Thread.sleep(10 * 1000);
+            elapsedTime = (new Date()).getTime() - startTime;
+        }
+
+        // Timeout, return false
+        return false;
+    }
+
+    public static void main(String argv[]) throws Exception {
+
+        // Get batch and storage account information from environment
+        String batchAccount = System.getenv("AZURE_BATCH_ACCOUNT");
+        String batchKey = System.getenv("AZURE_BATCH_ACCESS_KEY");
+        String batchUri = System.getenv("AZURE_BATCH_ENDPOINT");
+
+        String storageAccountName = System.getenv("STORAGE_ACCOUNT_NAME");
+        String storageAccountKey = System.getenv("STORAGE_ACCOUNT_KEY");
+
+        Boolean shouldDeleteContainer = true;
+        Boolean shouldDeleteJob = true;
+        Boolean shouldDeletePool = false;
+
+        Duration TASK_COMPLETE_TIMEOUT = Duration.ofMinutes(10);
+        String STANDARD_CONSOLE_OUTPUT_FILENAME = "stdout.txt";
+
+        // Create batch client
+        BatchSharedKeyCredentials cred = new BatchSharedKeyCredentials(batchUri, batchAccount, batchKey);
+        BatchClient client = BatchClient.Open(cred);
+
+        // Create storage container
+        CloudBlobContainer container = createBlobContainer(storageAccountName, storageAccountKey);
+
+        String userName = System.getProperty("user.name");
+        String poolId = userName + "-pooltest";
+        String jobId = "HelloWorldJob-" + userName + "-" + (new Date()).toString().replace(' ', '-').replace(':', '-').replace('.', '-');
+
+        try
+        {
+            CloudPool sharedPool = CreatePool(client, poolId);
+            submitJobAndAddTask(client, container, sharedPool.getId(), jobId);
+            if (waitForTasksToComplete(client, jobId, TASK_COMPLETE_TIMEOUT)) {
+                // Get the task command output file
+                CloudTask task = client.getTaskOperations().getTask(jobId, "mytask");
+
+                InputStream stream = client.getFileOperations().getFileFromTask(jobId, task.getId(), STANDARD_CONSOLE_OUTPUT_FILENAME);
+                String fileContent = IOUtils.toString(stream, "UTF-8");
+                System.out.println(fileContent);
+            }
+            else {
+                throw new TimeoutException("Task wasn't completed at time");
```

Comment:
I think this should read something more like "Task did not complete within the specified timeout" and maybe even include what the timeout was (i.e. `TASK_COMPLETE_TIMEOUT`)